spacy/util.py

```diff
@@ -1736,3 +1737,40 @@ def all_equal(iterable):
     (or if the input is an empty sequence), False otherwise."""
     g = itertools.groupby(iterable)
     return next(g, True) and not next(g, False)
+
+
+def is_port_in_use(port, host="localhost"):
+    """Check if localhost:port is in use."""
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.bind((host, port))
+        return False
+    except socket.error:
+        return True
+    finally:
+        s.close()
+
+
+def find_available_port(start, host, auto_select=False):
```

Comment:
Please add types for arguments and return values and fully specify the docstring.

Reply:
Done.